### ISSUE TITLE
Duplicated SNAPSHOT dependencies in war

### ISSUE THREAD

**Issue by zivkovic** [spring-projects/spring-boot#7743]
Hello!

In current version "1.4.3.RELEASE" of "spring-boot-maven-plugin" we have found a problem when packaging third party SNAPSHOT dependencies into war. Every SNAPSHOT dependency is duplicated (with SNAPSHOT and timestamp suffix in filename). We believe the cause of this bug is commit "Use standard file name for Maven dependencies" (commit number: "c51d836a152839075671f6f1254beb19f0bee9a4"). It works if we use "jar" packaging, but not when using "war". I have attached a small example demonstrating this problem.

```
$ unzip -l target/test-1.0.0-SNAPSHOT.war
Archive:  target/test-1.0.0-SNAPSHOT.war
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  23.12.2016 15:25   META-INF/
      311  23.12.2016 15:25   META-INF/MANIFEST.MF
        0  23.12.2016 15:25   WEB-INF/
        0  23.12.2016 15:25   WEB-INF/classes/
        0  23.12.2016 15:25   WEB-INF/lib/
      355  23.12.2016 15:25   WEB-INF/classes/Test.class
    61829  27.05.2016 12:53   WEB-INF/lib/commons-logging-1.2.jar
  1113497  23.12.2016 15:15   WEB-INF/lib/spring-core-4.3.6.BUILD-SNAPSHOT.jar
  1113497  22.12.2016 22:45   WEB-INF/lib/spring-core-4.3.6.BUILD-20161222.224510-3.jar
        0  23.12.2016 15:25   META-INF/maven/
        0  23.12.2016 15:25   META-INF/maven/com.test/
        0  23.12.2016 15:25   META-INF/maven/com.test/test/
     1676  23.12.2016 15:25   META-INF/maven/com.test/test/pom.xml
      118  23.12.2016 15:25   META-INF/maven/com.test/test/pom.properties
        0  23.12.2016 15:25   org/
```
As you can see in the directory structure above, there are two "spring-core" jar files inside of example war file.

This is a major problem for us and everyone else using the latest version of this plugin.

Best regards, 
Jan Živković

P.S. Current fix we are using is using this plugin with earlier version "1.4.2.RELEASE" which doesn't have this problem.

Example maven project zip file (couldn't upload to GitHub): https://mega.nz/#!mE1RlabQ!Ep1cNZRe8Ne6onHkQvfiW8RjfJO2x3uGbaVveuccr5g

**Comment by snicoll** [comment]
So the war plugin stores such snapshot without the timestamps and we do. That vaguely rings a bell, I need to remember what I did in `maven-war-plugin` now.

**Comment by snicoll** [comment]
@zivkovic thanks for the report, that one is pretty nasty. And thank you for the sample!

There is a fix available in `1.4.4.BUILD-SNAPSHOT` once the build has completed. Can you please give that a try on your project and report if you find anything suspicious still? Thanks a lot.

**Comment by zivkovic** [comment]
The fix is working, thank you.
